pkg/fleet/installer/packages/apminject/file.go

```diff
@@ -111,7 +111,7 @@ func (ft *fileMutator) mutate(ctx context.Context) (rollback func() error, err e
 	// validate final file if validation function provided
 	if ft.validateFinal != nil {
 		if err = ft.validateFinal(); err != nil {
-			if err := rollback(); err != nil {
+			if err = rollback(); err != nil {
 				log.Errorf("could not rollback file %s: %s", ft.path, err)
 			}
 			return nil, err
```

Comment:
@bmermet  which `err` value do you want to be returned here?

Reply:
I'm not familiar with this code, I think ideally we would want a new error that wraps the validate and the rollback errors to keep as much information as possible.
@raphaelgavache since you wrote this function, do you have an opinion?

Reply:
It should be the error of the validation that triggers the rollback. that rollback error is silenced with log and doesn't explained why we abort injection